### DISCUSSION TITLE
handling of ipv6 addresses clarification

### DISCUSSION THREAD

**Discussion by gekigek99** [gekigek99/minecraft-vanilla-server-hibernation#70] [Q&A] [answered]
Hi @lubocode! I saw this [2 lines](https://github.com/gekigek99/minecraft-server-hibernation/commit/85c94a227a4d819321e085793b8d13412536e4e5#diff-f84d989af5d52e959d2ae1f0cd483b363b69bb1d8430ec3aa05e1f32099682c1R173) last day and could not really interpret what it does... What was your goal when you wrote that?

[this](https://play.golang.org/p/A1ZzvK4jwcz) is a go playground example and the result does not seem very useful but maybe I'm missing something... 

for example, what does `clientSocket.RemoteAddr().String()` looks like? is it always the same? (i mean, there are a lot of possible ways to write an ipv6 address)

**Comment by lubocode** [comment, marked as answer]
Well, I assumed the writing of the IPv6 address as [IPv6address]:port, and as such, one would get the port and IPv6 address separated, no matter how the address itself is written. This is the way I have learned to use ports with IPv6 addresses, but if you know another, then please go ahead and mention it, so that those lines can be changed accordingly.

An example address would be `[2001:db8:85a3:8d3:1319:8a2e:370:7348]:443` **_(Taken from [Wikipedia](https://en.wikipedia.org/wiki/IPv6_address#Literal_IPv6_addresses_in_network_resource_identifiers). No idea where that leads^^)_**

**Reply by lubocode** [reply]
[Here ](https://play.golang.org/p/PtNEZUek1X-)is your example with the MC hibernate port attached.

**Comment by lubocode** [comment]
This solution also takes into account IPv4 addresses which will only have one `:` and as such that will be the last occurrence, right before the port number. 

Maybe adding a comment along the lines of `Split IP address and port number by means of locating last : which is the separator commonly used.` would sum it up most accurately.

**Reply by gekigek99** [reply]
ok thanks for the clarification
